src/test/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSTaxInfoTest.java

```diff
@@ -0,0 +1,64 @@
+package org.broadinstitute.hellbender.tools.spark.pathseq;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class PSTaxInfoTest {
+
+    @Test
+    public void testEqualsAndHashcode() {
+        final PSTaxInfo infoA = new PSTaxInfo();
+        infoA.parent_tax = "1";
+        infoA.length = 1000;
+        infoA.name = "tax A";
+        infoA.rank = "species";
+        infoA.ref_names.add("ref|NC_A1");
+        infoA.ref_names.add("ref|NC_A2");
+
+        final PSTaxInfo infoB = new PSTaxInfo();
+        infoB.parent_tax = "2";
+        infoB.length = 1000;
+        infoB.name = "tax B";
+        infoB.rank = "species";
+        infoB.ref_names.add("ref|NC_B1");
+
+        Assert.assertNotEquals(infoA, infoB);
+        Assert.assertNotEquals(infoA.hashCode(), infoB.hashCode());
+
+        final PSTaxInfo infoA2 = new PSTaxInfo();
+        infoA2.parent_tax = "1";
+        infoA2.length = 1000;
+        infoA2.name = "tax A";
+        infoA2.rank = "species";
+        infoA2.ref_names.add("ref|NC_A2");
+        infoA2.ref_names.add("ref|NC_A1");
+
+        Assert.assertEquals(infoA, infoA2);
+        Assert.assertNotEquals(infoA.hashCode(), infoA2.hashCode());
```

Comment:
In general the contract for `hashCode` should be that if the objects are equal according to `equals`, their hash codes should be the same. If you ever try to put these objects into a set or as keys in a map you might run into trouble. I missed this when I looked at the class but your test here points it out!

Do you really need to preserve the order of the ref names in the PSTaxInfo object? You could get rid of this problem if you put them into a set. If you really need them to preserve the order but want equals not to care about the order, you could sort them or put them into a set when you compute the hash code, although that would be inefficient.